### DISCUSSION TITLE
Solar panel generation-adding to energy dashboard

### DISCUSSION THREAD

**Discussion by usbrit** [hultenvp/solis-sensor#146] [Q&A]
I :can't seem to find the correct sensor in the energy dashboard link to add solar.

I think I need:
-My Solis Inverter AC Output Total Power
which when I monitor manually looks correct.
is it possible to add this to the energy dashboard sensor pull-down menu?

Thanks,
David

**Comment by hultenvp** [comment]
For the energy dashboard you'll need one of the energy sensors. Energy today had the best resolution (0.1 kWh steps). The others all have a resolution of 1kWh and are less suited for the energy dashboard.

**Comment by usbrit** [comment]
This is what I have on the dashboard:
![image](https://user-images.githubusercontent.com/53425475/186755662-4512d915-1bac-4f76-a5a2-7dec3de64c19.png)

Is this what you suggest for solar?:
![image](https://user-images.githubusercontent.com/53425475/186756035-20ded36e-868f-4720-9b25-61509fb6e97f.png)

This is what I was trying to use to give actual instantaneous solar output, but it's not on the list
![image](https://user-images.githubusercontent.com/53425475/186757424-263b8ed1-a2f2-4c40-967e-eeb539c1a543.png)

**Reply by hultenvp** [reply]
No the sensor is called energy today, so in your case My Solis Inverter Energy Today
<img width="666" alt="image" src="https://user-images.githubusercontent.com/61835400/186759493-3718c9c6-8b51-49e1-ab77-857b672a1915.png">

**Reply by usbrit** [reply]
OK got that, thanks.
How about the "Energy output to grid"?
I'm using "My Solis Inverter Daily On-grid Energy" but has some weird values tonight. Will leave it as is and check after a full day tomorrow.
![image](https://user-images.githubusercontent.com/53425475/186765850-d5a37797-ebae-4613-90aa-1b6c68627382.png)

Appreciate the assistance. I'll leave all sensor the same for 24 hours and then check the plots. This might be able to help with the other question I posted about inactive sensors.

**Reply by hultenvp** [reply]
The mapping of grid sensors might still be a bit ambiguous: https://github.com/hultenvp/solis-sensor/discussions/71#discussioncomment-2071111
I do not have a hybrid myself, so I miss the grid sensors, but looking back at how we originally mapped I'd expect 'Daily Grid Energy Used' to be more the total energy usage of your house today and 'Daily Grid Energy Purchased' more how much you drew from the grid today.

**Comment by drsmarsden** [comment]
![Screenshot 2022-08-26 at 11 19 31](https://user-images.githubusercontent.com/91994286/186883358-7c086d77-82c8-497a-97a8-f47ad1598244.png)
![Screenshot 2022-08-26 at 11 20 20](https://user-images.githubusercontent.com/91994286/186883376-d6f1ed25-f8dc-47b7-a488-3c9f4e7dfe6e.png)

This works for me, I have not taken the latest version yet due to other commitments

**Reply by usbrit** [reply]
My setup is now exactly the same and worked all day!
![image](https://user-images.githubusercontent.com/53425475/186961547-8e82f522-c83b-4988-9a13-545d5564af71.png)
The problem is that the export energy is no where near right. See below (today was a good day!)
![image](https://user-images.githubusercontent.com/53425475/186961130-7476ee9c-64e9-4072-bdd1-36e3e4921d33.png)

@drsmarsden is yours matching the Solis numbers?

**Comment by drsmarsden** [comment]
On a nice sunny day I can run on battery till the solar takes over and also recharges the batteries
(unless I plug in the car, like the last hour in the graph)

I spent a lot of effort minimising the steady state energy consumption , having run a business from home it turned out to be all IT !  now just POE switch , router and a few wifi AP !

**Reply by hultenvp** [reply]
Nice!
No EV here yet, but we'll be getting a heatpump in Q1. I'm now net producer over the year (currently +1.5MWh, hope to end at +1 end of year), but I'll become net consumer again once the heatpump is there.

**Comment by usbrit** [comment]
I think I have found the sensors required to calculate true "Output to Grid"
![image](https://user-images.githubusercontent.com/53425475/187061252-96067f01-a070-4911-936b-a8d5ae5d2bb5.png)

"Power Grid Total Power" shows the output values (I know this as I have my inverter set to discharge to the grid at 17.30-19.00 each evening at peak rate), but also shows the input from the grid as a negative.
"Inverter Total Consumption Power" shows the in value only

So the sum of these two sensor should be the "Output to Grid". Is there a way of doing this calculation and then creating a virtual sensor that could be used in the Energy Dashboard?

Thanks

**Comment by hultenvp** [comment]
A virtual sensor should work, see https://community.home-assistant.io/t/calculate-sum-two-entities-to-one-sensor/182063/6 for some tips.
You might need to give it the right state class of "measurement". 

If it really works for you I can consider offering it as a sensor in the integration. I'd recommend you create an improvement once it has been validated to work.
